rotkehlchen/data/importer.py

```diff
@@ -1014,3 +1015,167 @@ def import_shapeshift_trades_csv(self, filepath: Path) -> Tuple[bool, str]:
+                        amount=destination_amount,
+                        rate=Price(origin_amount / destination_amount),
+                        fee=Fee(fee),
+                        fee_currency=fee_asset,
+                        link='',
+                        notes=notes,
+                    )
+                    self.db.add_trades([trade])
+        # Outgoing
+        if origin == 'uphold' and destination != 'uphold' and transaction_type == 'out':
+            # Withdrawal
+            if destination_asset == origin_asset:
+                asset_movement = AssetMovement(
+                    location=Location.UPHOLD,
+                    category=AssetMovementCategory.WITHDRAWAL,
+                    address=None,
+                    transaction_id=None,
+                    timestamp=timestamp,
+                    asset=origin_asset,
+                    amount=origin_amount,
+                    fee=Fee(fee),
+                    fee_asset=fee_asset,
+                    link='',
+                )
+                self.db.add_asset_movements([asset_movement])
+            # Trades (sell)
+            else:
+                trade = Trade(
+                    timestamp=timestamp,
+                    location=Location.UPHOLD,
+                    base_asset=origin_asset,
+                    quote_asset=destination_asset,
+                    trade_type=TradeType.SELL,
+                    amount=origin_amount,
+                    rate=Price(destination_amount / origin_amount),
+                    fee=Fee(fee),
+                    fee_currency=fee_asset,
+                    link='',
+                    notes=notes,
+                )
+                log.debug(f'{trade.fee} {trade.rate}')
+                self.db.add_trades([trade])
+        # Incoming
+        if origin != 'uphold' and destination == 'uphold' and transaction_type == 'in':
+            # Deposit
+            if destination_asset == origin_asset:
+                asset_movement = AssetMovement(
+                    location=Location.UPHOLD,
+                    category=AssetMovementCategory.DEPOSIT,
+                    address=None,
+                    transaction_id=None,
+                    timestamp=timestamp,
+                    asset=origin_asset,
+                    amount=origin_amount,
+                    fee=Fee(FVal('0.0')),
+                    fee_asset=origin_asset,
+                    link='',
+                )
+                self.db.add_asset_movements([asset_movement])
+            # Trades (buy)
+            else:
+                trade = Trade(
+                    timestamp=timestamp,
+                    location=Location.UPHOLD,
+                    base_asset=destination_asset,
+                    quote_asset=origin_asset,
+                    trade_type=TradeType.BUY,
+                    amount=destination_amount,
+                    rate=Price(origin_amount / destination_amount),
```

Comment:
If `destination_amount` is ZERO here then you will get a division by zero error and the whole task will die. Please make sure it's not zero in this case and if it is skip accordingly perhaps with a log error.